### ISSUE TITLE
Exception Armrest::Error: #<Net::HTTPUnauthorized:0x0000561554b473e0> when executing terraspace up demo -y

### ISSUE THREAD

**Issue by teddimitrov** [boltops-tools/terraspace#297]
## Checklist

- [ x ] Upgrade Terraspace: Are you using the latest version of Terraspace? This allows Terraspace to fix issues fast. There's an Upgrading Guide: https://terraspace.cloud/docs/misc/upgrading/
- [ x ] Reproducibility: Are you reporting a bug others will be able to reproduce and not asking a question. If you're unsure or want to ask a question, do so on https://community.boltops.com
- [  ] Code sample: Have you put together a code sample to reproduce the issue and make it available?  Code samples help speed up fixes dramatically. If it's an easily reproducible issue, then code samples are not needed. If you're unsure, please include a code sample.

## My Environment

| Software         | Version |
| ---------------- | ------- |
| Operating System | Linux  5.15.90.1-microsoft-standard-WSL2  (Debian 11 Bullseye)    |
| Terraform        |  v1.4.2       |
| Terraspace       |   2.2.3      |
| Ruby             |  ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]        |

---

After executing terraspace up demo -y for the terraform code to be deployed in the azure subscription

Building .terraspace-cache/eastus/dev/stacks/demo
Current directory: .terraspace-cache/eastus/dev/stacks/demo
Creating Resource Group dev-eastus...
Exception Armrest::Error: #<Net::HTTPUnauthorized:0x0000561554b473e0>

download and install terraspace on debian
run terraspace new project infra --plugin azurerm --examples
followed by  terraspace up demo -y

According to bundle exec ruby azure_check.rb:

Successfully connected to Azure with your ARM_CLIENT_ID and ARM_CLIENT_SECRET

terraspace setup check:

Detected Terraspace version: 2.2.3
Detected Terraform bin: /usr/bin/terraform
Detected Terraform v1.4.2
Terraspace requires Terraform v0.12.x and above
You're all set!

**Comment by teddimitrov** [comment]
With the help of  "export ARMREST_LOG_LEVEL=debug" (from the post about similar issue #220 ) I can glean additional information and it seems related to authorization. I'll revisit the documentation and videos on the site to see if I can get past that error:

D, [2023-03-30T14:59:05.430528 #19537] DEBUG -- : resp #<Net::HTTPUnauthorized:0x0000564c41b0f1b0>
D, [2023-03-30T14:59:05.430654 #19537] DEBUG -- : resp.code 401
D, [2023-03-30T14:59:05.430688 #19537] DEBUG -- : resp.body {"error":{"code":"AuthenticationFailed","message":"Authentication failed. The 'Authorization' header is missing."}}

**Comment by teddimitrov** [comment]
After exporting again the azure environment variables (ARM_SUBSCRIPTION_ID and the rest) I was able to successfully run terraspace! Nothing wrong with terraspace itself but a newbie user error :)

PS. After I resolved the initial error I ran into a second roadblock but thanks to the output from "export ARMREST_LOG_LEVEL=debug" I was able to determine that it was due to a policy I had applied to the subscription - "Public network access should be disabled for PaaS services".
